libbeat/processors/add_locale/add_locale.go

```diff
@@ -1,29 +1,99 @@
 package actions
 
 import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
 	"time"
 
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/processors"
 )
 
-type addLocale struct{}
+type addLocale struct {
+	TimezoneFormat TimezoneFormat
+}
+
+type TimezoneFormat int
+
+const (
+	Abbrevation TimezoneFormat = iota
+	Offset
+)
+
+var timezoneFormats = [...]string{
+	"abbrevation",
+	"offset",
+}
+
+func (t TimezoneFormat) String() string {
+	return timezoneFormats[t]
+}
 
 func init() {
 	processors.RegisterPlugin("add_locale", newAddLocale)
 }
 
 func newAddLocale(c common.Config) (processors.Processor, error) {
-	return addLocale{}, nil
+	config := struct {
+		Format string `config:"format" validate:"required"`
+	}{}
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unpack the include_fields configuration: %s", err)
+	}
+
+	loc := addLocale{}
+
+	switch strings.ToLower(config.Format) {
+	case "abbrevation":
+		loc.TimezoneFormat = Abbrevation
+	case "offset":
+		loc.TimezoneFormat = Offset
+	default:
+		return nil, errors.New("given format is not valid")
+
+	}
+	return loc, nil
 }
 
 func (l addLocale) Run(event common.MapStr) (common.MapStr, error) {
-	zone, _ := time.Now().Zone()
-	event.Put("beat.timezone", zone)
+	format := l.Format()
+	event.Put("beat.timezone", format)
 
 	return event, nil
 }
 
+func (l addLocale) Format() string {
+	tm := time.Now()
+	var fmt string
+	switch l.TimezoneFormat {
+	case Abbrevation:
+		fmt, _ = tm.Zone()
+	case Offset:
+		//loc, _ := time.LoadLocation("America/Belize")
+		_, offset := tm.Zone()
+
+		offset = offset / 3600
+
+		if offset < 0 {
+			offset = offset - (offset * 2)
+			fmt += "-"
+		} else {
+			fmt += "+"
+		}
+
+		if offset < 10 {
+			fmt += "0"
+		}
+
+		fmt += strconv.Itoa(offset)
+		fmt += ":00"
```

Comment:
Timezone offsets do not all fall on one hour boundaries. See https://en.wikipedia.org/wiki/List_of_tz_database_time_zones

Here's an example calculation: https://play.golang.org/p/iuiuSxMZ1Q

This will need some test cases.